test/sharness/t0100-name.sh

```diff
@@ -225,6 +225,15 @@ test_name_with_self() {
         grep "argument \"ipfs-path\" is required" curl_out
         '
 
+        # test publishing with TTL
+        test_expect_success "'ipfs name publish --ttl=30m' succeeds" '
```

Comment:
Mind adding negative test that  uses different peerid and fails validation?
It could use `--enc=json`, that way we would have test for both types of output for free.